File: pt5/1198.find-smallest-common-element-in-all-rows.go
package main

func smallestCommonElement(mat [][]int) int {
	counts := map[int]int{}
	for _, row := range mat {
		for _, value := range row {
			counts[value]++
		}
	}

	for _, value := range mat[0] {
		if counts[value] == len(mat) {
			return value
		}
	}
	return -1
}

File: pt5/1198.find-smallest-common-element-in-all-rows.java
import java.util.*;

class Solution {
    public int smallestCommonElement(int[][] mat) {
        Map<Integer, Integer> count = new HashMap<>();
        for (int[] row : mat) {
            for (int value : row) {
                count.put(value, count.getOrDefault(value, 0) + 1);
            }
        }

        for (int value : mat[0]) {
            if (count.get(value) == mat.length) {
                return value;
            }
        }
        return -1;
    }
}

